test/Generics/protocol_where_clause.swift

```diff
@@ -65,3 +65,12 @@ struct BadConcreteNestedConforms: NestedConforms {
   // expected-error@-1 {{type 'ConcreteParentNonFoo2.T' (aka 'Float') does not conform to protocol 'Foo2'}}
   typealias U = ConcreteParentNonFoo2
 }
+
+// <rdar://problem/31041997> Some where-clause requirements aren't checked on conforming types
+protocol P1 {}
+protocol P2 { associatedtype T }
+protocol P3 { associatedtype U }
+protocol P4: P3 where U: P2, U.T: P1 {}
```

Comment:
Hm, isn't this isomorphic to the `NestedConforms` test above? @slavapestov's targeted fix https://github.com/apple/swift/pull/8428 that made that work includes a "rip this out once conformance access paths are plumbed through" comment, can that be executed now?